src/Firebase/Database/SseClient.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Kreait\Firebase\Database;
+
+use GuzzleHttp\Client;
+use Kreait\Firebase\Exception\ApiException;
+
+/**
+ * SSE Client to retrieve data from streaming rest API
+ */
+class SseClient
+{
+    const RETRY_DEFAULT_MS = 3000;
+    const END_OF_MESSAGE = "/\r\n\r\n|\n\n|\r\r/";
+
+    /**
+     * @var Client
+     */
+    private $apiClient;
+
+    /**
+     *
+     * @var GuzzleHttp\Psr7\Response
+     */
+    private $response;
+
+    /**
+     * @var string
+     */
+    private $url;
+
+    /**
+     * @var string - last received message id
+     */
+    private $lastId;
+
+    /**
+     * @var int - reconnection time in milliseconds
+     */
+    private $retry = self::RETRY_DEFAULT_MS;
+
+    /**
+     * @param ApiClient $apiClient
+     * @param string    $url
+     */
+    public function __construct(ApiClient $apiClient, string $url)
+    {
+        $this->apiClient = $apiClient;
+        $this->url = $url;
+
+        $this->connect();
+    }
+
+    /**
+     * Returns generator that yields new event when it's available on stream.
+     *
+     * @return Event[]
```

Comment:
`@return \Generator|ServerSentEvent[]` - the first is the actual return value, the second helps us developers to loop around ^^